collector/queries_test.go

```diff
@@ -1,198 +1,399 @@
-// Copyright 2025 Grafana Labs
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//      http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
 package collector
 
 import (
 	"strings"
 	"testing"
+	"time"
 )
 
-func TestQueries_NotEmpty(t *testing.T) {
-	queries := []struct {
-		name  string
-		query string
+// TestDurationToSQLInterval tests the duration to SQL interval conversion function.
+// Databricks SQL accepts both singular and plural forms (e.g., "1 HOUR" and "1 HOURS"),
+// but we use grammatically correct forms for clarity.
+func TestDurationToSQLInterval(t *testing.T) {
+	tests := []struct {
+		name     string
+		duration time.Duration
+		expected string
 	}{
-		{"billingDBUsQuery", billingDBUsQuery},
-		{"billingCostEstimateQuery", billingCostEstimateQuery},
-		{"priceChangeEventsQuery", priceChangeEventsQuery},
-		{"jobRunsQuery", jobRunsQuery},
-		{"jobRunStatusQuery", jobRunStatusQuery},
-		{"jobRunDurationQuery", jobRunDurationQuery},
-		{"taskRetriesQuery", taskRetriesQuery},
-		{"jobSLAMissQuery", jobSLAMissQuery},
-		{"pipelineRunsQuery", pipelineRunsQuery},
-		{"pipelineRunStatusQuery", pipelineRunStatusQuery},
-		{"pipelineRunDurationQuery", pipelineRunDurationQuery},
-		{"pipelineRetryEventsQuery", pipelineRetryEventsQuery},
-		{"pipelineFreshnessLagQuery", pipelineFreshnessLagQuery},
-		{"queriesQuery", queriesQuery},
-		{"queryErrorsQuery", queryErrorsQuery},
-		{"queryDurationQuery", queryDurationQuery},
-		{"queriesRunningQuery", queriesRunningQuery},
+		// Days - singular and plural
```

Comment:
You mention in the function description on the actual implementation that:
24 = 1 DAY
48h = 2 DAYS

Could you add tests for this too?

Similarly are we also needing to account for Months/years, etc?